### DISCUSSION TITLE
Derived model sharing the same table with its parent

### DISCUSSION THREAD

**Discussion by HKGx** [tortoise/tortoise-orm#1095] [Q&A]
In PonyORM when you try deriving models you get a single table.
Example PonyORM model:
```py
class Stage(db.Entity):
    id = PrimaryKey(int, auto=True)
    needed_messages = Required(int)
    message = Required(str)


class GiveawayStage(Stage):
    winnners = Required(int)
```
It then creates following table
```sql
CREATE TABLE "Stage" (
  "id" INTEGER PRIMARY KEY AUTOINCREMENT,
  "needed_messages" INTEGER NOT NULL,
  "message" TEXT NOT NULL,
  "classtype" TEXT NOT NULL,
  "winnners" INTEGER
)
```

In Tortoise when you derive a model from another, you get a completely new table. Is there a way to obtain the same behaviour as with PonyORM? Or do I have to do it manually by creating a tagged model with enum field?

**Comment by kfields** [comment]
Try this:

```py
class AbstractStage(db.Entity):
    id = PrimaryKey(int, auto=True)
    needed_messages = Required(int)
    message = Required(str)

    class Meta:
        abstract = True

    def __str__(self):
        return self.name

class GiveawayStage(AbstractStage):
    winners = Required(int)
```

This works the same way as Django.  It won't create a table for abstract classes.  The derived class will use the inherited properties to create a table.

**Reply by HKGx** [reply]
but I don't want Stage to be abstract, that's the point. I want two models use one table for inheritance reasons.

**Reply by Tobi-De** [reply]
> but I don't want Stage to be abstract, that's the point. I want two models use one table for inheritance reasons.

Hi, did you tried ? I don't have an answer to your problem but just curious to known what happened, did the code broke ? If so, what error message did you receive?

**Reply by HKGx** [reply]
@Tobi-De I just created a single model with all the necessary fields, that also holds an enum field. 
It's not great but *it just works*.
https://github.com/HKGx/lambo/blob/pre-db/lambo/models/activity_tracker_model.py#L19-L33

**Reply by Tobi-De** [reply]
> @Tobi-De I just created a single model with all the necessary fields, that also holds an enum field. It's not great but _it just works_. [HKGx/lambo@`pre-db`/lambo/models/activity_tracker_model.py#L19-L33](https://github.com/HKGx/lambo/blob/pre-db/lambo/models/activity_tracker_model.py#L19-L33)

Ok then
